### PR TITLE
876 - Manage status bar appearance in ODS theme

### LINKED ISSUE #876: Manage system bars colors in OdsTheme

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: theme-innovation-cup/src/main/java/com/orange/ods/theme/innovationcup/InnovationCupColors.kt [hunk 1]
```diff
@@ -43,9 +44,7 @@ internal val InnovationCupLightColors = OdsColors(
         alert = Yellow
     ),
 
-    components = OdsComponentsColors(
-        systemBarsBackground = PrimaryDark
```
Comment (florentmaitre) [inline theme-innovation-cup/src/main/java/com/orange/ods/theme/innovationcup/InnovationCupColors.kt:47]: Why did you remove the `PrimaryDark` color? Maybe I'm wrong but I would have thought the equivalent code would be this:

```kotlin
statusBar = OdsStatusBarColors(
    background = PrimaryDark,
    isAppearanceLight = false
)
```

File: theme-contract/src/main/java/com/orange/ods/theme/colors/OdsComponentsColors.kt [hunk 2]
```diff
@@ -52,10 +68,40 @@ class OdsComponentsColors(
...
+@Suppress("FunctionName")
+@Deprecated(
+    "Please use OdsComponentColors function with only init parameter instead.",
+    replaceWith = ReplaceWith("OdsComponentsColors(init)")
+)
 fun OdsComponentsColors(systemBarsBackground: Color, init: OdsComponentsColors.Builder.() -> Unit = {}): OdsComponentsColors.Builder {
-    return OdsComponentsColors.Builder(systemBarsBackground).apply(init)
+    return OdsComponentsColors.Builder().apply(init).apply {
+        statusBar = OdsStatusBarColors(background = systemBarsBackground)
+    }
```
Comment (florentmaitre) [inline theme-contract/src/main/java/com/orange/ods/theme/colors/OdsComponentsColors.kt:84]: This can be written with only one apply:

```kotlin
return OdsComponentsColors.Builder().apply {
    init()
    statusBar = OdsStatusBarColors(background = systemBarsBackground)
}
```

File: theme-contract/src/main/java/com/orange/ods/theme/colors/OdsComponentsColors.kt [hunk 3]
```diff
@@ -52,10 +68,40 @@ class OdsComponentsColors(
...
+ * Customizable colors for status bar
+ *
+ * @attribute background Background color of the status bar.
+ * @attribute isAppearanceLight Controls the items color of the status bar. When `true`, the items on the bar can be read clearly on a light background.
+ *
+ * Use [Colors.DefaultOdsStatusBarColors] to get the default component's colors based on Material Colors defined in theme
+ */
+data class OdsStatusBarColors(
+    val background: Color,
+    val isAppearanceLight: Boolean = false
```
Comment (florentmaitre) [inline theme-contract/src/main/java/com/orange/ods/theme/colors/OdsComponentsColors.kt:97]: I wonder if `isAppearanceLight` should be optional, because there is no logical default value (for instance I would set it to `true` because I mainly use light theme on my phone 😄)

Whatever we do, don't forget to update `OdsStatusBarColors(systemBarsBackground, false)` line 47 if needed.

Comment (paulinea) [reply theme-contract/src/main/java/com/orange/ods/theme/colors/OdsComponentsColors.kt:97]: I set `isAppearanceLight` to `false` because this is the default appearance of a status bar in Material 2. Even if you are in light theme on your phone, by default the app has a primary dark background status bar: [https://developer.android.com/develop/ui/views/theming/themes](https://developer.android.com/develop/ui/views/theming/themes) (This is maybe not the case in M3). There is an option to change this behavior to have a light status bar in the SDK but this is not the default behavior.

File: theme-innovation-cup/src/main/java/com/orange/ods/theme/innovationcup/InnovationCupColors.kt (hunk 1 above)
Comment (paulinea) [reply theme-innovation-cup/src/main/java/com/orange/ods/theme/innovationcup/InnovationCupColors.kt:47]: I wonder if it was good before because, if I use `PrimaryDark` here, everything is in black and white except the status bars. It's a bit weird but maybe it's normal?

**Review by florentmaitre — APPROVED** [review]
